control-plane/agents/common/src/msg_translation/v0.rs

```diff
@@ -11,10 +11,18 @@ use common_lib::{
     },
 };
 
-use crate::msg_translation::{AgentToIoEngine, IoEngineToAgent, TryIoEngineToAgent};
+use crate::msg_translation::{IoEngineToAgent, TryIoEngineToAgent};
 use rpc::io_engine as v0_rpc;
 use std::convert::TryFrom;
 
+/// Trait for converting agent messages to io-engine messages.
```

Comment:
I don't understand this move, why do we need separate traits, which are essentially the same?

Reply:
Since `AgentToIoEngine` has the `IoEngineMessage` type embeded in it, the conversion from v0 is in conflict with v1, because the trait cannot be implemented on the same transport type twice.

Reply:
Ah right... Could instead have?
```rust
/// Trait for converting agent messages to io-engine messages.
pub trait AgentToIoEngine<IoEngineMessage> {
    /// Conversion of agent message to io-engine message.
    fn to_rpc(&self) -> IoEngineMessage;
}
```
But that would require a lot of changes, so yeah let's keep it the way you have it now

Reply:
Yeah, but yeah, this would be a good refactor. Will take it up later, as this is blocking @Ab-hishek for now.